lib/queue.js

```diff
@@ -1,110 +1,108 @@
 'use strict';
 
-const metasync = {};
-module.exports = metasync;
+module.exports = (api) => {
 
-metasync.ConcurrentQueue = function(
-  // ConcurrentQueue
-  concurrency, // number of simultaneous and asynchronously executing tasks
-  timeout // process timeout (optional), for single item
-) {
-  this.isOnPause = false;
-  this.concurrency = concurrency;
-  this.timeout = timeout;
-  this.count = 0;
-  this.items = [];
-  this.events = {
-    error: null,
-    timeout: null,
-    empty: null,
-    process: null
+  api.metasync.ConcurrentQueue = function(
+    // ConcurrentQueue
+    concurrency, // number of simultaneous and asynchronously executing tasks
+    timeout // process timeout (optional), for single item
+  ) {
+    this.isOnPause = false;
+    this.concurrency = concurrency;
+    this.timeout = timeout;
+    this.count = 0;
+    this.items = [];
+    this.events = {
+      error: null,
+      timeout: null,
+      empty: null,
+      process: null
+    };
   };
-};
 
-metasync.ConcurrentQueue.prototype.add = function(
-  item // add item to queue
-) {
-  if (!this.isOnPause) {
-    if (this.count < this.concurrency) {
-      this.next(item);
-    } else {
-      this.items.push(item);
+  api.metasync.ConcurrentQueue.prototype.add = function(
+    item // add item to queue
+  ) {
+    if (!this.isOnPause) {
+      if (this.count < this.concurrency) this.next(item);
+      else this.items.push(item);
     }
-  }
-};
+  };
 
-metasync.ConcurrentQueue.prototype.next = function(
-  item // process next item from queue
-) {
-  const queue = this;
-  let timer;
-  if (queue.isOnPause) return;
-  queue.count++;
-  if (queue.timeout) {
-    timer = setTimeout(() => {
-      const err = new Error('ConcurrentQueue timed out');
-      queue.emit('timeout', err);
-    }, queue.timeout);
-  }
-  const stub = (item, callback) => callback();
-  const fn = queue.events.process || stub;
-  fn(item, () => {
-    queue.count--;
+  api.metasync.ConcurrentQueue.prototype.next = function(
+    item // process next item from queue
+  ) {
+    const queue = this;
```

Comment:
Arrow functions capture `this` lexically, there's no need in copying the reference to a new variable.